src/handlers/gestures/GestureDetector.tsx

```diff
@@ -79,20 +79,35 @@ function dropHandlers(preparedGesture: GestureConfigReference) {
   }
 }
 
+function checkGestureCallbacksForWorklets(gesture: GestureType) {
+  // if a gesture is explicitly marked to run on the JS thread there is no need to check
+  // if callbacks are worklets as the user is aware they will be ran on the JS thread
+  if (gesture.config.runOnJS) {
+    return;
+  }
+
+  const areAllWorklets = !gesture.handlers.isWorklet.includes(false);
+  const areAllNotWorklets = !gesture.handlers.isWorklet.includes(true);
```

Comment:
<small>small</small> nit: maybe using `some` and `every` is more descriptive

Reply:
Bigger nit: you use them only in negation, maybe it's better to use `someAreWorklets`, `someAreCallbacks`.

Reply:
Changed in https://github.com/software-mansion/react-native-gesture-handler/pull/1862/commits/f6af3008e67af7e85deb4ca81858a282c093bada.